core/templates/domain/exploration/AnswerGroupObjectFactory.ts

```diff
@@ -90,7 +90,7 @@ export class AnswerGroupObjectFactory {
 
   createFromBackendDict(
       answerGroupBackendDict: AnswerGroupBackendDict,
-      interactionId: string | null
+      interactionId: string,
```

Comment:
Is it guaranteed that the backend representation of interactionId is non-null? e.g. when there is no interactionId present in a state.

Reply:
> Is it guaranteed that the backend representation of interactionId is non-null? e.g. when there is no interactionId present in a state.

https://github.com/oppia/oppia/blob/c6a5b7abf9a095cb4a785c6cc8b1b4ad1ecc4756/core/domain/state_domain_test.py#L334
I think backend test cannot return None type id, so id will not be null after created from backend dict.

Reply:
Actually, that line is just for that particular test, to confirm that the id is not None in that case (to help the type-checker). But in general it is possible for the ID to be None in some cases. For example: try creating a new exploration and saving it before selecting an interaction for the first state.

So this will need some analysis and reading of the code -- we should remove null from the parts that cannot be null, and keep that option for the parts that can be.  Presumably in this case, when creating an Answer Group object, the interaction ID would already exist, because you can't have an answer group without an interaction. But please double-check that you are not disallowing null in places where it should not be disallowed.

Reply:
> Actually, that line is just for that particular test, to confirm that the id is not None in that case (to help the type-checker). But in general it is possible for the ID to be None in some cases. For example: try creating a new exploration and saving it before selecting an interaction for the first state.
> 
> So this will need some analysis and reading of the code -- we should remove null from the parts that cannot be null, and keep that option for the parts that can be. Presumably in this case, when creating an Answer Group object, the interaction ID would already exist, because you can't have an answer group without an interaction. But please double-check that you are not disallowing null in places where it should not be disallowed.

How can I address some test pass null interactionId to answerGroup, can I delete or change the tests directly?

Reply:
Yes, if the test describes an impossible condition, you can modify the test to make it legitimate.

Reply:
@462702985 Could you please confirm the answer to Kevin's original question, i.e. "Is it guaranteed that the backend representation of interactionId is non-null? e.g. when there is no interactionId present in a state."? Would be good to confirm that and the chain of reasoning that leads to that conclusion.

Reply:
> @462702985 Could you please confirm the answer to Kevin's original question, i.e. "Is it guaranteed that the backend representation of interactionId is non-null? e.g. when there is no interactionId present in a state."? Would be good to confirm that and the chain of reasoning that leads to that conclusion.

When we create a defaultState, this interactionId is null before saving a state.

Reply:
Yup, but how about other cases? E.g. when the interactionId is not null but then later is changed to null because e.g. the interaction is deleted?

Reply:
https://github.com/oppia/oppia/blob/4a738cb0bde277f265affa2f2cbd3885b3b358c3/core/templates/domain/state/StateObjectFactory.ts#L139
Here we create a state where interactionId is null.
https://github.com/oppia/oppia/blob/4a738cb0bde277f265affa2f2cbd3885b3b358c3/core/templates/domain/exploration/StatesObjectFactory.ts#L76
when we delete interaction, this setState will set interactionId to be null again.